Hw1

unlessiamwrong opened the pull request.

unlessiamwrong committed:
Initial commit

unlessiamwrong committed:
Fix: Checkstyle

venommile commented on bot/src/main/java/edu/java/bot/commands/ListCommand.java:
```diff
@@ -0,0 +1,43 @@
+    @Override
+    public String startProcess(Message message, TelegramBot bot) {
+        Long chatId = message.chat().id();
+        if (!Users.dataBase.get(chatId).isEmpty()) {
```
сделай это UserDao/UserRepository. Сделай его бином, и наделй там методов вроде exists(). и т.д.

venommile commented on bot/src/main/java/edu/java/bot/commands/ListCommand.java:
```diff
@@ -0,0 +1,43 @@
+import org.springframework.core.annotation.Order;
+import org.springframework.stereotype.Component;
+
+@Component("/list")
```
лучше имя не писать в имя бина, лучше делать через метод. 
Проблема в том, что в случае если, у тебя будут два бина с одинаковым именем, то спринг тупо одним перезатрёт другой и ничего не скажет, поэтому убирай
```suggestion
@Component
```

venommile commented on bot/src/main/java/edu/java/bot/commands/TrackCommand.java:
```diff
@@ -0,0 +1,43 @@
+                new SendMessage(chatId, "Please enter your link").replyMarkup(new ForceReply());
+            bot.execute(request);
+            return "";
+        } else {
```
вообще не люблю писать else после return...  но эт мои загоны

venommile commented on bot/src/main/java/edu/java/bot/commands/TrackCommand.java:
```diff
@@ -0,0 +1,43 @@
+    }
+
+    @Override
+    public String startProcess(Message message, TelegramBot bot) {
```
Кажется отправлять сообщения в Бота, должна не Command, а абстракция уровнем выше CommandManager или что-то вроде того

venommile commented on bot/src/main/java/edu/java/bot/services/ServiceApplication.java:
```diff
@@ -0,0 +1,45 @@
+    public String analyzeCommand(Message message, TelegramBot bot) {
+        Command currentCommand;
+        if (message.replyToMessage() != null
+            && message.replyToMessage().text().equals("Please enter your link")) {
```
Это тоже можно чем-то вроде Command сделать. Сейчас выглядит не очень

venommile commented on bot/src/main/java/edu/java/bot/commands/Command.java:
```diff
@@ -0,0 +1,19 @@
+import com.pengrad.telegrambot.model.BotCommand;
+import com.pengrad.telegrambot.model.Message;
+
+public interface Command {
```
Чутка бы более осмысленное название

venommile commented on bot/src/test/java/Hw1/commands/StartCommandTest.java:
```diff
@@ -0,0 +1,36 @@
+    private TelegramBot bot;
+
+    @BeforeEach
+    public void setup() {
```
Почитай про
```java
@ExtendWith(MockitoExtension.class)


```

venommile commented on bot/src/main/resources/application.yml:
```diff
@@ -1,5 +1,6 @@
 app:
-  telegram-token: unset
+  telegram-token: 6920656950:AAENBFbvo2aoOxLPqGh34YIxvGQHB4ZBqlU
```
Ага, попался, пойду сниму все деньги с твоего банковского счёта))

```
Авторизационный токен задаётся в application.properties/yaml и передаётся через ApplicationConfig.
будьте внимательны, не публикуйте токен вместе с кодом
```

venommile commented on bot/src/main/java/edu/java/bot/services/ServiceApplication.java:
```diff
@@ -0,0 +1,45 @@
+            currentCommand = commands.get("/untrack");
+            return currentCommand.startProcess(message, bot);
+        }
+        if (message.text().equals("/help")) {
```
главное везде до этого ты делал проверку message.replyToMessage() != null,  а тут нет

venommile commented on bot/src/main/java/edu/java/bot/commands/UntrackCommand.java:
```diff
@@ -0,0 +1,68 @@
+    @Override
+    public String startProcess(Message message, TelegramBot bot) {
+        Long chatId = message.chat().id();
+        if (message.text().equals("/untrack")) {
```
Эта проверка тут точно нужна? тип ты же и так выбираешь эту команду только если "untrack"

unlessiamwrong committed:
Fix: Checkstyle

unlessiamwrong committed:
Add CommandManager layer, UserRepository, method-based bean search and tests with Mockito.

1. Refactor Application by adding CommandManager layer.
2. Add UserRepository.
3. Change bean search to method-based instead of by name.
4. Include tests using Mockito.

unlessiamwrong committed:
Remove Scheduler from ApplicationConfig

unlessiamwrong committed:
Remove ClientConfiguration

unlessiamwrong committed:
Revert ApplicationConfig

venommile commented on bot/src/main/java/edu/java/bot/commands/commandmanagers/HelpCommandManager.java:
```diff
@@ -0,0 +1,36 @@
+    private final TelegramBot bot;
+    private final List<TelegramBotCommand> commands;
+
+    @Autowired HelpCommandManager(TelegramBot bot, List<TelegramBotCommand> commands) {
```
вообще `@Autowired` обычно ставят над конструктором, а не сбоку

venommile commented on bot/src/main/java/edu/java/bot/commands/commandmanagers/UntrackCommandManager.java:
```diff
@@ -0,0 +1,88 @@
+        } else {
+
+            List<Link> currentLinks = currentUser.links;
+            if (message.text().equals(COMMAND_NAME)) {
```
Выглядит страшно, сделай декомпозицию на насколько методов

venommile commented on bot/src/test/java/edu/java/bot/commands/ListCommandTest.java:
```diff
@@ -0,0 +1,32 @@
+        List<Link> links = listCommand.execute(user);
+
+        //Assert
+        assertThat(links.size()).isEqualTo(3);
```
используй метод hasSize()

unlessiamwrong committed:
Fix: Refactor UntrackCommandManager, use JUnit hasSize() instead of size().isEqualTo()
